Sources/StandardLib.swift

```diff
@@ -43,10 +43,12 @@ extension Bool: Decodable {
 }
 
 internal func castOrFail<T>(e: Extractor) throws -> T {
-    let rawValue = e.rawValue
+    return try castOrFail(e.rawValue)
+}
 
-    guard let result = rawValue as? T else {
-        throw typeMismatch("\(T.self)", actual: rawValue, keyPath: nil)
+internal func castOrFail<T>(any: Any?) throws -> T {
```

Comment:
I'm not sure what the real world use case is. The users can implement `decode(e: Extractor)` by using required initializer or also can use `Transformer` API for decoding non-final classes. Honestly I'm not willing to make it `public`.